tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -682,6 +686,164 @@ public void AppWithResources (ApplePlatform platform, string runtimeIdentifiers)
+				file.CopyTo (tempPath, false);
+			}
+			foreach (DirectoryInfo subDir in dir.GetDirectories ()){
+				CopyDirectoryContents (subDir, Path.Combine (DestDirectory, subDir.Name));
+			}
+		}
+
+		JsonDocument ProcessAssets (string assetsPath) {
+			var output = new StringBuilder ();
+			var executable = "xcrun";
+			var arguments = new string [] { "--sdk", "iphoneos", "assetutil", "--info", assetsPath };
+			var rv = Execution.RunWithStringBuildersAsync (executable, arguments, standardOutput: output, standardError: output, timeout: TimeSpan.FromSeconds (120)).Result;
+			Assert.AreEqual (0, rv.ExitCode, "ExitCode");
+
+			// This Execution call produces an output with an objc warning. We just want the json below it.
+			if (output.ToString ().StartsWith ("objc"))
+				output.Remove (0, output.ToString ().IndexOf ('\n'));
+
+			return JsonDocument.Parse (output.ToString ());
+		}
+
+		HashSet<string> FindAssets (JsonDocument doc)
+		{
+			var jsonArray = doc.RootElement.EnumerateArray ();
+			var foundElements = new HashSet<string> ();
+
+			foreach (var item in jsonArray) {
+				var result = GetTarget (item);
+				if (result is not null)
+					foundElements.Add (result);
+			}
+			return foundElements;
+		}
+
+		string? GetTarget (JsonElement item) {
+			if (item.TryGetProperty ("AssetType", out var assetType)) {
+				foreach (var target in XCAssetTargets) {
+					var result = GetTarget (item, assetType, target);
+					if (result is not null)
+						return result;
+				}
+			}
+			return null;
+		}
+
+		string? GetTarget (JsonElement item, JsonElement assetType, XCAssetTarget target)
+		{
+			if (assetType.ToString () == target.AssetType) {
+				if (item.TryGetProperty (target.CategoryName, out var value)) {
+					if (target.Values.Contains (value.ToString ()))
+						return string.Concat (assetType.ToString (), ".", value.ToString ());
+				}
+			}
+			return null;
+		}
+
+		class XCAssetTarget {
+			public string AssetType { get; set; }
+			public string CategoryName { get; set; }
+			public string [] Values { get; set; }
+			public XCAssetTarget (string assetType, string categoryName, string[] values) {
+				AssetType = assetType;
+				CategoryName = categoryName;
+				Values = values;
+			}
+		}
+
+		static XCAssetTarget[] XCAssetTargets = {
+			new XCAssetTarget ("Image", "RenditionName", new string [] { "samplejpeg.jpeg", "samplejpg.jpg",
+				"samplepdf.pdf", "samplepng2.png", "spritejpeg.jpeg", "loopsvg.svg" }),
+
+			new XCAssetTarget ("Data", "Name", new string [] { "BmpImageDataTest", "JsonDataTest", "DS_StoreDataTest",
+				"DngImageDataTest", "EpsImageDataTest", "TiffImageDataTest" }),
+
+			new XCAssetTarget ("Color", "Name", new string [] { "ColorTest" }),
+
+			new XCAssetTarget ("Contents", "Name", new string [] { "SpritesTest" }),
+
+			new XCAssetTarget ("Texture Rendition", "Name", new string [] { "TextureTest" }),
+
+			new XCAssetTarget ("Vector", "RenditionName", new string [] { "samplepdf.pdf", "loopsvg.svg" }),
```

Comment:
Lay manual would re-write this to type less due to his fat fingers, no need to add the class fin the new, it can be inferred, same for the array.

Reply:
The XCAssetTarget was fine being left off, but the string [] complained when removed!